internal/controller/isbservicerollout_controller.go

```diff
@@ -317,6 +327,7 @@ func (r *ISBServiceRolloutReconciler) processExistingISBService(ctx context.Cont
 		}
 
 		isbServiceRollout.Status.MarkDeployed(isbServiceRollout.Generation)
+		r.customMetrics.ReconciliationDuration.WithLabelValues(ControllerISBSVCRollout, "update").Observe(time.Since(syncStartTime).Seconds())
```

Comment:
unfortunately, we don't know that this is an "update" vs a "no-op", do we?

Reply:
similarly here, `if isbServiceNeedsUpdating {...}`

Reply:
nit: sorry, but what do you think about enclosing it in the `else` block that's directly above it? I think then it's very clear that this statement goes with the `dataLossPrevention=false` case only and there will be less chance of messing it up later. Otherwise, I had to look closely at the `dataLossPrevention=true` code above to confirm that it was in fact returning and would't hit this.